apps/solaredgemonitor/solaredge_monitor.star

```diff
@@ -5,167 +5,671 @@ Description: Energy production and consumption monitor for your SolarEdge solar
     api_key = config.str("api_key")
     site_id = humanize.url_encode(config.str("site_id", ""))
+    tz = config.get("$tz", "Europe/Zurich")
+    has_battery = False  #  assume no battery until we have data
 
     if api_key and site_id:
         url = URL.format(site_id)
-        data = cache.get(url)
-        if not data:
-            rep = http.get(url, params = {"api_key": api_key})
-            if rep.status_code != 200:
-                fail("SolarEdge API request failed with status {}".format(rep.status_code))
-            data = rep.body()
-
-            # TODO: Determine if this cache call can be converted to the new HTTP cache.
-            cache.set(url, data, ttl_seconds = CACHE_TTL)
-        o = json.decode(data)
+        rep = http.get(url, params = {"api_key": api_key}, ttl_seconds = CACHE_TTL)
+        if rep.status_code != 200:
+            print(rep.body())
+            return render_fail(rep)
+        o = json.decode(rep.body())
+
+        consumption = 0
+        production = 0
+        if config.bool("show_summary", False) == True:
+            today_energy = get_energy_details(site_id, api_key, tz, "day")
+            if today_energy:
+                for m in today_energy["meters"]:
+                    if m["type"] == "Consumption":
+                        for v in m["values"]:
+                            consumption += v["value"]
+                    elif m["type"] == "Production":
+                        for v in m["values"]:
+                            production += v["value"]
+
+        if config.bool("show_autarky", False) == True:
+            autarky_day = get_autarky_percent(site_id, api_key, tz, "day")
+            autarky_24h = get_autarky_percent(site_id, api_key, tz, "24h")
+            autarky_month = get_autarky_percent(site_id, api_key, tz, "month")
+            autarky_year = get_autarky_percent(site_id, api_key, tz, "year")
+        else:
+            autarky_day = None
+            autarky_24h = None
+            autarky_month = None
+            autarky_year = None
     else:
+        print("using dummy data")
         o = DUMMY_DATA
+        consumption = 3141
+        production = 6282
+        autarky_day = 99
+        autarky_24h = 97
+        autarky_month = 95
+        autarky_year = 76
+    frames = []
 
     o = o["siteCurrentPowerFlow"]
     unit = o["unit"]
     connections = o["connections"]
-    points = []
-    flows = []
-    if o["PV"]:
-        img = SOLAR_PANEL if o["PV"]["status"] == "Active" else SOLAR_PANEL_OFF
-        points.append((img, o["PV"]["currentPower"]))
-        if {"from": "PV", "to": "Load"} in connections:
-            dir = 1
-        else:
-            dir = 0
-        flows.append(dir)
-    if o["LOAD"]:
-        points.append((HOUSE, o["LOAD"]["currentPower"]))
-        if {"from": "GRID", "to": "Load"} in connections:
-            dir = -1
-        elif {"from": "LOAD", "to": "Grid"} in connections:
-            dir = 1
+    if o["STORAGE"]:
```

Comment:
This crashes on installations without a battery. Use `if "STORAGE" in o` instead (also throughout the file)

Reply:
Done.